pkg/config/tmpl/rulesconfig.go

```diff
@@ -169,6 +171,14 @@ func (rc *RulesConfig) Merge(other TemplateConfig) error {
 				return err
 			}
 		}
+		// Add the Name field if we're creating a rule (keyPrefix starts with "RulesBasedSampler.Rules." but doesn't contain "Conditions")
+		if strings.HasPrefix(keyPrefix, "RulesBasedSampler.Rules.") && !strings.Contains(keyPrefix, "Conditions") {
```

Comment:
not a blocker, but it would be good to have this in a separate function to reuse it in both places where this logic is needed

Reply:
I made it better. It doesn't remove all the duplication but when I tried to, it looked worse.